modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java

```diff
@@ -156,4 +160,12 @@ protected String getMimeType()
         return MimeTypes.XML;
     }
 
+    @Override
+    protected void addInInterceptorCorrectingInterceptors(ServerFactoryBean sfb)
+    {
+        if (isProxy() && isProxyEnvelope() && CxfUtils.interceptorOfClassIsPresent(sfb.getInInterceptors(), SAAJInInterceptor.class))
```

Comment:
Is this only needed for the case of SAAJInInterceptor.

Reply:
As far I looked into it, yes. Either way, the interceptor check if in the CXF's message contents there is a SOAPMessage, from which it regenerates the xml stream with the envelope